hibernate-core/src/test/java/org/hibernate/orm/test/jpa/criteria/TreatPathTest.java

```diff
@@ -129,13 +131,17 @@ public void testTreatAnyValue(EntityManagerFactoryScope scope) {
 	}
 
 	private void testCriteriaTreat(EntityManager entityManager, String property, Object value) {
-		CriteriaBuilder cb = entityManager.getCriteriaBuilder();
-		CriteriaQuery<Linkage> criteria = cb.createQuery( Linkage.class );
-		Root<Linkage> root = criteria.from( Linkage.class );
-		Path<LocalTerm> asLocalTerm = cb.treat( root.get( "term" ), LocalTerm.class );
-		Predicate predicate;
-		if ( value instanceof Collection<?> ) {
-			predicate = asLocalTerm.get( property ).in( value );
+		testCriteriaTreat( entityManager, property, value, false );
+	}
+
+	private void testCriteriaTreat(EntityManager entityManager, String property, Object value, boolean useIn) {
+		final CriteriaBuilder cb = entityManager.getCriteriaBuilder();
+		final CriteriaQuery<Linkage> criteria = cb.createQuery( Linkage.class );
+		final Root<Linkage> root = criteria.from( Linkage.class );
+		final Path<LocalTerm> asLocalTerm = cb.treat( root.get( "term" ), LocalTerm.class );
+		final Predicate predicate;
+		if ( useIn ) {
+			predicate = cb.literal( value ).in( asLocalTerm.get( property ) );
```

Comment:
This refactoring kind of seems wrong. You switched from `asLocalTerm.get( property ).in( value )` to `cb.literal( value ).in( asLocalTerm.get( property ) )`, which is reverse AFAIU.

Reply:
That was on purpose, I think the way I wrote the test before was wrong. This case with plural values is a bit flaky anyway - I thought about disabling it but for the purpose of checking treat itself it was enough to expect an `UsupportedOperationException`.

Reply:
Fixed the test now.